model/src/main/java/uk/gov/pay/commons/api/json/MicrosecondPrecisionDateTimeDeserializer.java

```diff
@@ -0,0 +1,34 @@
+package uk.gov.pay.commons.api.json;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.PropertyNamingStrategy;
+import com.fasterxml.jackson.databind.annotation.JsonNaming;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+
+@JsonNaming(PropertyNamingStrategy.SnakeCaseStrategy.class)
+public class MicrosecondPrecisionDateTimeDeserializer extends StdDeserializer<ZonedDateTime> {
```

Comment:
Few tests would be nice to guard for any changes

Reply:
added some tests